cmd/ovsdb-mon/shell.go

```diff
@@ -338,3 +318,148 @@ func colordiff(a, b interface{}) string {
 	}
 	return strings.TrimRight(buf.String(), "\n")
 }
+
+// filterAPI returns the conditional API that filters based on the provided filter expression
+// Expression is [FIELD]=[VALUE]
+func (s *OvsdbShell) filterAPI(tableName string, expr string) (client.ConditionalAPI, error) {
+	condModel, err := s.dbModel.NewModel(tableName)
+	if err != nil {
+		return nil, err
+	}
+	parts := strings.Split(expr, "=")
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("Invalid filter expression: %s", expr)
+	}
+
+	field := parts[0]
+	value := parts[1]
+
+	fieldVal := reflect.ValueOf(condModel).Elem().FieldByName(field)
+	if !fieldVal.IsValid() {
+		return nil, fmt.Errorf("field %s not present in database table %s", field, tableName)
+	}
+	if fieldVal.Kind() != reflect.String {
+		return nil, fmt.Errorf("filters only support string values")
+	}
+
+	fieldVal.Set(reflect.ValueOf(value))
+
+	return s.ovs.Where(condModel), nil
+}
+
+// listAutoComplete returns the list of strings to use for list command auto-completion
+func (s *OvsdbShell) listAutoComplete(tableName string, prefix string, args []string) []string {
+	autocomplete := []string{}
+	// Find where "--filter" is in the args
+	filterIndex := -1
+	for i, arg := range args {
+		if arg == "--filter" {
+			filterIndex = i
+			break
+		}
+	}
+	switch filterIndex {
+	case -1: // --filter has not been used yet, offer it
+		autocomplete = append(s.tableFields[tableName], "--filter")
+	case len(args) - 1: // We're autocompeting a filter, return the indexes and append "="
+		for _, f := range s.indexes[tableName] {
+			autocomplete = append(autocomplete, f+"=")
+		}
+	default: // filter has already been processed
+		autocomplete = s.tableFields[tableName]
+	}
+	if prefix != "" {
+		autocomplete = addLower(autocomplete)
+	}
+	return autocomplete
+}
+
+// exactFieldName returns the exact field name based on a field name that might have different capitalization
+func (s *OvsdbShell) exactFieldName(tableName string, field string) (string, bool) {
+	stype := s.dbModel.Types()[tableName].Elem()
+	for i := 0; i < stype.NumField(); i++ {
+		if strings.EqualFold(stype.Field(i).Name, field) {
+			return stype.Field(i).Name, true
+		}
+	}
+	return "", false
+}
+
+func (s *OvsdbShell) listTable(tableName string, c *ishell.Context) {
+	columns := []string{}
+	var filter string
+	var err error
+	isFilter := false
+	for _, arg := range c.Args {
+		if arg == "--filter" {
+			isFilter = true
+		} else {
+			if isFilter {
+				if filter != "" {
+					c.Println("Only one --filter statement allowed")
+					return
+				}
+				filter = arg
+				isFilter = false
+			} else {
+				if col, exists := s.exactFieldName(tableName, arg); exists {
+					columns = append(columns, col)
+				} else {
+					c.Println("Field %s not found in table %s", arg, tableName)
```

Comment:
c.Println("Field", arg, "not found in table", tableName)

of maybe use Printf?

```
ovsdb-mon.nb list Logical_Switch_Port --help
2021/11/07 23:15:53 libovsdb: trying to connect to DB unix:///run/openvswitch/ovnnb_db.sock
2021/11/07 23:15:53 libovsdb: connected to unix:/run/openvswitch/ovnnb_db.sock
OVSDB Monitoring Shell
Field %s not found in table %s --help Logical_Switch_Port
```

Reply:
Good catch, thanks!